pkg/transport/generic_transport_controller.go

```diff
@@ -366,13 +396,22 @@ func (c *genericTransportController) updateWrappedObjectsAndFinalizer(ctx contex
 		return fmt.Errorf("failed to get current wrapped objects that are owned by Binding '%s' - %w", binding.GetName(), err)
 	}
 	// calculate desired state
-	desiredWrappedObject, err := c.initializeWrappedObject(ctx, binding)
+	destToDesiredWrappedObject, bindingErrors, err := c.initializeWrappedObject(ctx, binding)
+	if err != nil {
+		return fmt.Errorf("failed to build wrapped object(s) from Binding '%s' - %w", binding.GetName(), err)
+	}
+	bindingCopy := binding.DeepCopy()
+	bindingCopy.Status = v1alpha1.BindingStatus{
+		ObservedGeneration: binding.Generation,
+		Errors:             bindingErrors,
+	}
+	_, err = c.bindingClient.UpdateStatus(ctx, bindingCopy, metav1.UpdateOptions{FieldManager: ControllerName})
 	if err != nil {
-		return fmt.Errorf("failed to build wrapped object from Binding '%s' - %w", binding.GetName(), err)
+		return err
```

Comment:
wrap err - 
```
return fmt.Errorf("failed to update status of Binding '%s' - %w", binding.GetName(), err)
```

Reply:
Does that wrap really add anything? This is already in the context of working on a Binding object. If the context's logger does not already have the Binding reference in its values, it should.

Reply:
yes, it actually does. it shows errors from all layers which helps in debugging the problem.

Reply:
Revised to wrap.